lib/full-site-editing/templates.php

```diff
@@ -61,6 +61,29 @@ function gutenberg_register_template_post_type() {
 }
 add_action( 'init', 'gutenberg_register_template_post_type' );
 
+/**
+ * Get the edit link for templates.
+ *
+ * @param string $link    The original link.
+ * @param int    $post_id The custom post id.
+ */
+function gutenberg_get_edit_template_link( $link, $post_id ) {
+	$templates = gutenberg_get_block_templates(
+		array( 'wp_id' => $post_id ),
+		'wp_template'
+	);
+	$template  = $templates ? $templates[0] : null;
```

Comment:
I would trust `$templates` less and check both if it is an array and if it has length > 1 and only then access the zero index, because if somehow it comes back from `gutenberg_get_block_templates` in any other way then this ternary is unreliable. Like:

```PHP
if ( ! is_array( $templates ) || count( $templates ) < 1 ) {
	return $link;
}
$template = $templates[0];
```

Reply:
Outdated :D